docs/how-to-guides/data-basics/4-building-a-create-page-form.md

```diff
@@ -0,0 +1,373 @@
+
+```js
+export function EditPageForm( { pageId, onCancel, onSaveFinished } ) {
+	// ...
+	return (
+		<PageForm
+			title={ page.title }
+			onChangeTitle={ handleChange }
+			hasEdits={ hasEdits }
+			lastError={ lastError }
+			isSaving={ isSaving }
+			onCancel={ onCancel }
+			onSave={ handleSave }
+		/>
+	);
+}
+
+export function PageForm( { title, onChangeTitle, hasEdits, lastError, isSaving, onCancel, onSave } ) {
+	return (
+		<div className="my-gutenberg-form">
+			<TextControl
+				label="Page title:"
+				value={ title }
+				onChange={ onChangeTitle }
+			/>
+			{ lastError ? (
+				<div className="form-error">Error: { lastError.message }</div>
+			) : (
+				false
+			) }
+			<div className="form-buttons">
+				<Button
+					onClick={ onSave }
+					variant="primary"
+					disabled={ ! hasEdits || isSaving }
+				>
+					{ isSaving ? <Spinner /> : 'Save' }
+				</Button>
+				<Button onClick={ onCancel } variant="tertiary">
+					Cancel
+				</Button>
+			</div>
+		</div>
+	);
+}
+```
+
+This code quality change should not change anything about how the application works. Let’s try to edit a page just to be sure:
+
+![](/docs/how-to-guides/data-basics/media/create-form/edit-page-form.png)
+
+Great! The edit form is still there, and now we have a building block to power the new `CreatePageForm`.
+
+### Step 3: Build a CreatePageForm
+
+The only thing that `CreatePageForm` component must do is to provide the following seven properties needed to render the `PageForm` component:
+
+* title
+* onChangeTitle
+* hasEdits
+* lastError
+* isSaving
+* onCancel
+* onSave
+
+Let’s see how we can do that:
+
+#### Title, onChangeTitle, hasEdits
+
+The `EditPageForm` updated and saved an existing entity record that lived in the Redux state. Because of that, we relied on the `editedEntityRecords` selector.
+
+In case of the `CreatePageForm`, however, there is no pre-existing entity record. There is only an empty form. Anything that the user types is local to that form, which means we can kep track of it using the React’s `useState` hook:
+
+```js
+export function CreatePageForm( { onCancel, onSaveFinished } ) {
+	const [ title, setTitle ] = useState();
+	const handleChange = ( title ) => setTitle( title );
+	return (
+		<PageForm
+			title={ title }
+			onChangeTitle={ setTitle }
+			hasEdits={ !! title }
+			{ /* ... */ }
+			/>
+	);
+}
+```
+
+#### onSave, onCancel
+
+In the `EditPageForm`, we dispatched the `saveEditedEntityRecord('postType', 'page', pageId )` action to save the edits that lived in the Redux state.
+
+In the `CreatePageForm`, however, we do not have any edits in the Redux state, nor we do have any `pageId`. The action we need to dispatch in this case is called [`saveEntityRecord`](https://developer.wordpress.org/block-editor/reference-guides/data/data-core/#saveentityrecord) (without the word _Edited_ in the name) and it accepts an object representing the new entity record instead of a `pageId`.
```

Comment:
```suggestion
In the `CreatePageForm` however, we do not have any edits in the Redux state, nor we do have a `pageId`. The action we need to dispatch in this case is called [`saveEntityRecord`](https://developer.wordpress.org/block-editor/reference-guides/data/data-core/#saveentityrecord) (without the word _Edited_ in the name) and it accepts an object representing the new entity record instead of a `pageId`.
```